### DISCUSSION TITLE
Memory optimisations

### DISCUSSION THREAD

**Discussion by fokion** [ovh/venom#701] [Ideas]
- Lazy loading executors , not keeping them in memory , maybe not requiring to have the field `executor` in order to define the name and maybe assume the name of the executor is the name of the file ( we reduce the need of having to parse the file more than once)
- Generate reporting for the tests during execution so that we dont have to keep a lot of variables in the memory till the end.

**Comment by fokion** [comment]
Happy to create tickets to get the ball rolling

**Comment by fokion** [comment]
https://github.com/ovh/venom/pull/704

**Comment by fokion** [comment]
#705

**Comment by fokion** [comment]
https://github.com/ovh/venom/pull/735

**Comment by ivan-velasco** [comment]
@fokion Did this ever get more traction? Is there any work left?

**Reply by fokion** [reply]
I am not contributing anymore in this project due to the fact that PRs are getting stale and have no traction. I opted on having a fork of this project with my updates.

**Comment by yesnault** [comment]
Hi,
Yes, there were many pull-request, some rejected, some merged, some wip.
There were many subjects, not only linked to memory optimization. If something can be improved (other than using flag to disable some features), we'll be happy to discuss about that in a dedicated thread.
